src/dask_awkward/layers/layers.py

```diff
@@ -31,20 +33,14 @@ def mock(self) -> AwkwardBlockwiseLayer:
         layer.__dict__.pop("_dims", None)
         return layer
 
-    def __repr__(self) -> str:
-        return "Awkward" + super().__repr__()
-
     def __getstate__(self) -> dict:
-        d = self.__dict__.copy()
-        import pickle
+        # Indicator that this layer has been serialised
+        state = self.__dict__.copy()
+        state["has_been_unpickled"] = True
+        return state
```

Comment:
I added this to detect when a layer has been unpickled, in order to disable column optimisation. Does this align with your expectations?

Reply:
I'll ping @martindurant since he wrote the original `__getstate__`

Reply:
I can't think of why we'd want to disable optimization if a layer has been pickled. Has this come up in discussion before?

Reply:
I don't know _why_ the `_meta` was stored in the first place — it doesn't seem like we were reading it anywhere. My interpretation was that we don't want to run column optimisation if we evaluate the graph on the scheduler because it would already have been optimised. I don't know how one would even enter into the column optimisation routines to begin with, though, so this interpretation is likely flawed.

Reply:
[`self._meta`](https://github.com/dask-contrib/dask-awkward/blob/5981337650231e5bde00bc0cf9c8b9358e86404c/src/dask_awkward/layers/layers.py#L76) is getting assigned right now in `AwkwardInputLayer` (which inherits from `AwkwardBlockwiseLayer`); we were using it to instantiate a new form, followed by a new typetracer array to run the typetracer only graph after a call of the `mock` method. I think Martin wrote the code to pop the _meta attribute upon serialization because TypetracerArray wasn't serializable. I think with the refactoring you've done we no longer need to worry about `_meta` being a part of the layer, correct?

Reply:
`_meta` can still be within the layer as an attribute of e.g. the `io_func`, but not directly in `self._meta` for the `AwkwardInputLayer`.